### PR TITLE
Coverage doctest fix

### LINKED ISSUE #16716: bin/coverage_doctest.py does not work
```pytb
$./bin/coverage_doctest.py
DOCTEST COVERAGE for /Users/aaronmeurer/Documents/Python/sympy/sympy/sympy
======================================================================

Traceback (most recent call last):
  File "./bin/coverage_doctest.py", line 601, in <module>
    no_color=args.no_color, sphinx=args.sphinx)
  File "./bin/coverage_doctest.py", line 541, in go
    verbose=verbose, no_color=no_color, exact=False, sphinx=sphinx)
  File "./bin/coverage_doctest.py", line 549, in go
    any(name in file for name in skip_paths))):
NameError: name 'skip_paths' is not defined
```

It works if you remove the skip_paths check, but we should look into the history of the file to see what that was, because clearly something broke at some point.

### PR DESCRIPTION
**Pull request by s-hanko**
#### References to other Issues or PRs
Fixes #16716

#### Brief description of what is fixed or changed
Coverage_doctest.py was failing due to name changes between python2 and python3; this pull request updates the names to the python3 versions. It also was not dealing with dynamic docstrings correctly; now it skips them. Variables also have been updated to be more descriptive. 

#### Release Notes

NO ENTRY

### CODE REVIEW

File: bin/coverage_doctest.py [hunk 1]
```diff
@@ -425,19 +429,19 @@ def coverage(module_path, verbose=False, no_color=False, sphinx=True):
...
     c_sph = []
 
     f_skipped = []
-    f_md = []
-    f_mdt = []
+    f_missing_doc = []
+    f_missing_doctest = []
     f_has_doctest = []
-    f_idt = []
+    f_indierect_doctest = []
```
Comment (asmeurer) [inline bin/coverage_doctest.py:444]: "indirect" is spelled wrong in these variable names.

**Comment by asmeurer** [comment]
I ran this and it no longer errors. The final result is

```
======================================================================
TOTAL DOCTEST SCORE for sympy: 83% (13821 of 16531)
TOTAL SPHINX SCORE for sympy: 0% (0 of 16531)
```

which is obviously wrong for the Sphinx score. So that still needs to be fixed.

**Comment by asmeurer** [comment]
Another good change we could make here would be to sort the modules when processing them, so that they are printed in order.

**Comment by oscarbenjamin** [comment]
I didn't try this out but the changes to the code in general look good.

**Comment by smichr** [comment]
So let's merge this and I will open an issue for the points @asmeurer raised.

**Pull request merged**